docs/Modern-PersistedQueries.md

```diff
@@ -0,0 +1,162 @@
+    
+    ```json
+    {
+      "3be4abb81fa595e25eb725b2c6a87508": "query TodoItemRefetchQuery(\n  $itemID: ID!\n) {\n  node(id: $itemID) {\n    ...TodoItem_item_2FOrhs\n  }\n}\n\nfragment TodoItem_item_2FOrhs on Todo {\n    text\n    isComplete\n}\n"
+    }
+    ```
+
+3. It also generates a complete query map file at `[your_src_dir]/queryMap.graphql.json`. This file contains all the query ids 
+and their operation texts. You can specify a custom file path for this file by using the `--persist-output` option:
+
+```js
+"scripts": {
+  "relay": "relay-compiler --src ./src --schema ./schema.graphql --persist --persist-output ./src/queryMaps/queryMap.desktop.graphql.json"
+}
+```
+
+The example above writes the complete query map file to `./src/queryMaps/queryMap.desktop.graphql.json`. You need to ensure all the directories
+leading to the `graphql.json` file exist. Also note that the file extension has to be `.json`.
+
+### Network layer changes
+You'll need to modify your network layer fetch implementation to pass a queryId parameter in the POST body instead of a query parameter:
+
+```js
+function fetchQuery(operation, variables,) {
+  return fetch('/graphql', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({
+      queryId: operation.id, // NOTE: pass md5 hash to the server
+      // query: operation.text, // this is now obsolete because text is null 
+      variables,
+    }),
+  }).then(response => {
+    return response.json();
+  });
+}
+```
+
+## Usage on the server
+On the server, you'll need to map the query id in the POST body to the real operation text. You can utilise the complete 
+`queryMap.graphql.json` file to do this so you'll need a copy of it on your server. 
+
+For universal applications where the client and server code are in one project, this is not an issue since you can place 
+the query map file in a common location accessible to both the client and the server.
+
+For applications where the client and server projects are separate, you can solve this by having an additional npm run script
+to push the query map to a location accessible by your server:
+
+```js
+"scripts": {
+  "push-queries": "node ./pushQueries.js",
+  "relay": "relay-compiler --src ./src --schema ./schema.graphql --persist && npm run push-queries"
+}
+```
+
+Some possibilities of what you can do in `./pushQueries.js`:
+
+* `git push` to your server repo
+
+* save the query maps to a database
+
+It is also possible to push your query maps to the server at runtime, without the server knowing the query ids at the start.
+The client optimistically sends a query id to the server, which does not have the query map. The server then in turn requests
+for the full query text from the client so it can cache the query map for subsequent requests. This is a more complex approach
+requiring the client and server to interact to exchange the query maps.
```

Comment:
Should this be another entry in the list of options, rather than a separate paragraph?

Reply:
I have re-organised the server push options to have a compile time and a run time section. Optimistic query is a run-time solution and so is separate from the `npm run` option which is a compile time solution. Please have a look and let me know if this looks ok. Thanks @alloy.